src/dsl_vlplot_function/dsl_vlplot_function.jl

```diff
@@ -0,0 +1,223 @@
+struct VLFrag
+    positional::Vector{Any}
+    named::Dict{String,Any}
+end
+
+function vlfrag(args...; kwargs...)
+    return VLFrag(Any[i for i in args], Dict{String,Any}(string(k)=>convert_nt_to_dict(v) for (k,v) in kwargs))
+end
+
+convert_nt_to_dict(item) = item
+
+function convert_nt_to_dict(item::NamedTuple)
+    return VLFrag(Any[], Dict{String,Any}(string(k)=>convert_nt_to_dict(v) for (k,v) in pairs(item)))
+end
+
+function convert_nt_to_dict(item::VLFrag)
+    return VLFrag(item.positional, Dict{String,Any}(string(k)=>convert_nt_to_dict(v) for (k,v) in pairs(item.named)) )
+end
```

Comment:
How about recurse into `Dict{<:Union{Symbol,AbstractString}}` as well?  ATM, `vlplot(mark=:point, encoding=Dict("x"=>Dict("field"=>:x,), :y=>Dict("field"=>:y,)))` does not seem to work.

Reply:
So right now the idea is that the fact that we use `Dict`s at all is completely an internal implementation detail. Both `vlplot` and `@vlplot` convert their respective inputs (`NamedTuple` and the `{}` syntax) right away into `VLFrag`, and then we can rely on the tree being composed of that. Only later (and entirely internally) do we convert some stuff to `Dict`.

That generally makes the code that does the adjustments a lot easier because we only have to deal with one time. I also kind of think it is good to keep the API surface area tight. What would we gain if we accept `Dict` there? So I'm tempted right now to leave as is...

Reply:
VegaLite.jl seems to be very transparent that underlying object model is "just JSON", which is great.  You can directly use vega-lite JS library manual as a reference thanks to this.  Given that, supporting any JSON-like object does not seem to be far fetched.

In fact, I think it's reasonable to support `vlplot(::AbstractDict)`.  This trivialize at least some part of #196 as you don't need any "VegaBase.jl" to create a vega-lite/vega object.  It's just a plain old dictionary.  So, you can create vega-lite specs or fragments of it without depending on any library.

> Both `vlplot` and `@vlplot` convert their respective inputs (`NamedTuple` and the `{}` syntax) right away into `VLFrag`, and then we can rely on the tree being composed of that.

I'm not arguing to use `Dict` as an intermediate representation.  If you want to use `VLFrag` uniformly, I think that's great.  I'm just suggesting to add something like

```julia
function convert_nt_to_dict(item::AbstractDict{<:Union{Symbol,AbstractString}})
    return VLFrag([], Dict{String,Any}(string(k)=>convert_nt_to_dict(v) for (k,v) in pairs(item)) )
end
```

Reply:
Alright, I see the argument, but I think I need to think a bit about it :)

In the meantime, one can always directly pass a dict to `VLSpec`, right? At least for experts that should cover some of the scenarios, right?